homeassistant/config_entries.py

```diff
@@ -640,7 +640,9 @@ async def _async_process_on_unload(self) -> None:
                 await asyncio.gather(*pending)
 
     @callback
-    def async_start_reauth(self, hass: HomeAssistant) -> None:
+    def async_start_reauth(
+        self, hass: HomeAssistant, context: dict[str, Any] | None = None
```

Comment:
Please add a unit test where we test that we can update the flow context.

Reply:
I haven't had time to dig in yet, but I can't see an existing test for async_start_reauth so I was hoping you could give me a pointer on how you think it should be structured.

I have tests in xiaomi_ble that cover this, but i'm not sure how to adapt them because I don't have an integration. They aren't really unittests I guess. Should I mock the call to hass.config_entries.flow.async_init and verify it is called correctly?

Reply:
Something like this but use the entry method to start the reauth flow instead of raising an exception.

https://github.com/home-assistant/core/blob/dev/tests/test_config_entries.py#L2863-L2896

Reply:
Cheers, that was a big help. aa8f9c49855ab28d69ed02955e864c7c975a997a adds a test based on that. It also covers not starting dupe flows.

Reply:
Perfect!